### PR TITLE
Add verify_ssl option to gluon.utils.download

### PR DESCRIPTION
**Pull request by leezu**
## Description ##
Sometimes datasets may be hosted on servers that serve invalid SSL certificates. If so, this allow disabling the SSL certificate verification. The default behavior is not changed.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] Add verify_ssl option to gluon.utils.download

### CODE REVIEW

**Comment by larroy** [comment]
LGTM

File: tests/python/unittest/test_gluon_utils.py [hunk 1]
```diff
@@ -31,4 +31,12 @@ def test_download_successful():
...
     mx.gluon.utils.download("https://raw.githubusercontent.com/apache/incubator-mxnet/master/README.md",
                             path=tmpfile)
-    assert os.path.getsize(tmpfile) > 100
\ No newline at end of file
+    assert os.path.getsize(tmpfile) > 100
+
+def test_download_ssl_verify():
+    with warnings.catch_warnings(record=True) as warnings_:
+        mx.gluon.utils.download(
+            "https://mxnet.incubator.apache.org/", verify_ssl=False)
```
Comment (larroy) [inline tests/python/unittest/test_gluon_utils.py]: I would mock the call to requests to avoid doing network IO on a unit test.

**Comment by larroy** [comment]
Is better to mock it, in python is super simple. Using localhost can also fail if the port is used, so best to avoid it. Let's not overcomplicate things. Mocking the request is easy:  

https://stackoverflow.com/questions/15753390/how-can-i-mock-requests-and-the-response

**Comment by leezu** [comment]
Thanks. I have now changed all `gluon.utils.download` tests to make use of Mocking (the two tests already present before this PR caused network access).

**Comment by leezu** [comment]
Where are test dependencies currently tracked? This PR would introduce `mock` as a new dependency. For linux it seems I could add `mock` the list of installed packages in the CI setup shell scripts, but I am not sure about windows.

In case there is no centralized place to declare test/development dependencies, would it make sense to track these in the `setup.py` script and adapt the CI scripts to install via `pip install -e '.[tests]'` to also install these dependencies?

**Comment by szha** [comment]
AFAIK windows CI slave images are still manually built. @marcoabreu might have update

**Comment by larroy** [comment]
We can workaround this by adding pip install mock after the activate call on the Jenkinsfile .  parametrization of windows host is starting now and will take a while.

**Pull request merged**